### PR TITLE
Social media milestone2

### PR DESCRIPTION
**Pull request by RNtaate**
## What does this PR do?
- [X] Creates a Friendship model.
- [X] Creates associations between the Friendship model and User model.
- [X] Modifies users index and show page to send and accept friend requests.

**Note:** 
1. All frienship features such as sending and accepting friend requests, unfriending have been added.
2. The friendship feature was fully achieved using a unique id column in the friendship database table called `friendship_id`, generated using the `friender_id` and `friendee_id`.

### CODE REVIEW

File: app/controllers/friendships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    redirect_back(fallback_location: root_path)
+  end
+
+  def destroy
+    @friendship = Friendship.find_by(friendship_id: params[:id])
+    @friendship.destroy
+
+    redirect_back(fallback_location: root_path)
+    # redirect_to users_path
```
Comment (MuriloRoque) [inline app/controllers/friendships_controller.rb]: - [x] Please remove commented pieces of code.

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Status: Changes requested :red_circle: 

### Hello @RNtaate and @ershadul1 :wave: 

### You have done a very good job in this milestone :clap: 

### We only have a few things to fix, please check my suggestions below:

- [x] I should only see posts from people I am friends with. Currently, I can see posts from any users 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)):

Screenshot from 2020-10-12 14-14-11

- [x] Please try adding a test for the suggestion above so you can see if it worked :smiley: .

- [x] Check my lone comment in your code below.

## Summary

- Your commit messages, README file and PR messages are very professional :clap: 
- Your code is clean and there's no logic in views :100: 
- Your features are working as intended, except for the suggestion I made :rocket: 

### Kindly make the requested changes and submit for another review, please :pray: 

### I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Comment by RNtaate** [comment]
## Changes made
- [X] Only friends can see each other's posts.
- [X] Added feature test for only viewing friends' posts.
- [X] Removed commented pieces of code.

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️

- Excellent work applying the necessary changes.

### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**